tests/test_zero_data_parallel/common.py

```diff
@@ -119,7 +120,7 @@ def check_params_padding(model, zero_model, loose=False):
 def check_sharded_params_padding(model, zero_model, loose=False):
     rank = dist.get_rank()
     for p, zero_p in zip(model.parameters(), zero_model.parameters()):
-        zero_p = zero_p.col_attr.sharded_data_tensor.payload.to(p.device).float()
+        zero_p = zero_p.data.to(p.device).float()
```

Comment:
If zero_p is a ShardedParameter, how could it has .data?
Or the name zero_model is wrong?

Reply:
`zero_p` is a `Parameter`. We set `p.data = p.col_attr.sharded_data_tensor.payload` in `optimizer.step()`. Actually, if we reuse fp16 shard, `p.col_attr.sharded_data_tensor.payload` in this function is grad instead of param.